plugins/printer/snmp__hpclj

```diff
@@ -0,0 +1,185 @@
+use constant oid_yellow_cur		=> ".1.3.6.1.2.1.43.11.1.1.9.1.4";
+use constant oid_tray1_max		=> ".1.3.6.1.2.1.43.8.2.1.9.1.1";
+use constant oid_tray1_cur		=> ".1.3.6.1.2.1.43.8.2.1.10.1.1";
+use constant oid_tray2_max		=> ".1.3.6.1.2.1.43.8.2.1.9.1.2";
+use constant oid_tray2_cur		=> ".1.3.6.1.2.1.43.8.2.1.10.1.2";
+use constant oid_tray3_max		=> ".1.3.6.1.2.1.43.8.2.1.9.1.3";
+use constant oid_tray3_cur		=> ".1.3.6.1.2.1.43.8.2.1.10.1.3";
+
+
+#####
+# Config
+###
+if (defined $ARGV[0] and $ARGV[0] eq "config") {
+	my ($host) = Munin::Plugin::SNMP->config_session();
+	print "host_name $host\n" unless $host eq 'localhost';
+	print "graph_title HP Printer Consumption Levels
+graph_info This graph shows Consumption-Levels on HP Printers.
+graph_vlabel %
+graph_args --upper-limit 100 -l 0
+graph_scale no
+";
+	print "black.label Black Toner Level
+black.draw LINE2
+black.type GAUGE
+black.colour 000000
+black.warning 10:
+black.critical 3:
+black.min 0
+black.max 100
+";
+	if(oidExists(oid_cyan_max)){
+		print "cyan.label Cyan Toner Level
+cyan.draw LINE2
+cyan.type GAUGE
+cyan.colour 00FFFF
+cyan.warning 5:
+cyan.critical 1:
+cyan.min 0
+cyan.max 100
+";
+	}
+	if(oidExists(oid_magenta_max)){
+		print "magenta.label Magenta Toner Level
+magenta.draw LINE2
+magenta.type GAUGE
+magenta.colour FF00FF
+magenta.warning 5:
+magenta.critical 1:
+magenta.min 0
+magenta.max 100
+";
+	}
+	if(oidExists(oid_yellow_max)){
+		print "yellow.label Yellow Toner Level
+yellow.draw LINE2
+yellow.type GAUGE
+yellow.colour FFFF00
+yellow.warning 5:
+yellow.critical 1:
+yellow.min 0
+yellow.max 100
+";
+	}
+	if(oidExists(oid_tray1_max)){
+		print "tray1.label Tray1 Fill Level
+tray1.draw LINE1
+tray1.type GAUGE
+tray1.colour 333333
+tray1.min 0
+tray1.max 100
+";
+	}
+	if(oidExists(oid_tray2_max)){
+		print "tray2.label Tray2 Fill Level
+tray2.draw LINE1
+tray2.type GAUGE
+tray2.colour 666666
+tray2.min 0
+tray2.max 100
+";
+	}
+	if(oidExists(oid_tray3_max)){
+		print "tray3.label Tray3 Fill Level
+tray3.draw LINE1
+tray3.type GAUGE
+tray3.colour 999999
+tray3.min 0
+tray3.max 100
+";
+	}
+	exit 0;
+}
+
+
+#####
+# Get Values
+###
+printValue("black",		oid_black_cur,		oid_black_max);
+printValue("cyan",		oid_cyan_cur,		oid_cyan_max);
+printValue("magenta",	oid_magenta_cur,	oid_magenta_max);
+printValue("yellow",	oid_yellow_cur,		oid_yellow_max);
+printValue("tray1",		oid_tray1_cur,		oid_tray1_max);
+printValue("tray2",		oid_tray2_cur,		oid_tray2_max);
+printValue("tray3",		oid_tray3_cur,		oid_tray3_max);
+
+
+#####
+# Subroutines
+###
+sub printValue {
+	if(not defined $_[0] || not defined $_[1] || not defined $_[2]) {
+		exit 0;
+	}
+	my $field = $_[0];
+	my $oid_cur = $_[1];
+	my $oid_max = $_[2];
+	
+	if(not oidExists($oid_cur) || not oidExists($oid_max)){
+		exit 0;
+	}
+	
+	my $val_max	= $session->get_single($oid_max) || 'U';
+	my $val_cur	= $session->get_single($oid_cur) || 'U';
+	if ($val_max ne 'U' && $val_cur ne 'U') {
```

Comment:
The problem here is that get_single() returns 'U' in case a '0' is received. I am not quite sure if this is a bug with the function or my implementation.

Either way, here we are assuming that an undefined value for the Max-Counter (which should always be >0) determines that the counter is not available, thus return no value.